# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 1de84c438bb9
# Full Hash: 1de84c438bb9a9c51992234ccbc747e58edac49a
# Author: Karl Tomlinson <karlt+@karlt.net>
# Date: 2020-12-17 09:27:48
# Description:
#   Bug 1673767 trap fatal X errors when looking up _NET_WM_NAME on _NET_SUPPORTING_WM_CHECK window r=stransky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D99855
# ==============================================================================

diff -r a14336cb528b -r 1de84c438bb9 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Wed Dec 16 18:42:35 2020 +0000
+++ b/widget/gtk/nsWindow.cpp	Wed Dec 16 19:01:46 2020 +0000
@@ -21,6 +21,7 @@
 #include "mozilla/TouchEvents.h"
 #include "mozilla/UniquePtrExtensions.h"
 #include "mozilla/WidgetUtils.h"
+#include "mozilla/X11Util.h"
 #include "mozilla/XREAppData.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/WheelEventBinding.h"
@@ -1922,13 +1923,17 @@
   if (!property || !req_type) {
     return false;
   }
-  result =
-      XGetWindowProperty(xdisplay, wmWindow, property,
-                         0L,         // offset
-                         INT32_MAX,  // length
-                         false,      // delete
-                         req_type, &actual_type_return, &actual_format_return,
-                         &nitems_return, &bytes_after_return, &prop_return);
+  {
+    // Suppress fatal errors for a missing window.
+    ScopedXErrorHandler handler;
+    result =
+        XGetWindowProperty(xdisplay, wmWindow, property,
+                           0L,         // offset
+                           INT32_MAX,  // length
+                           false,      // delete
+                           req_type, &actual_type_return, &actual_format_return,
+                           &nitems_return, &bytes_after_return, &prop_return);
+  }
 
   if (result != Success || bytes_after_return != 0) {
     return false;
